File: Algorithms/Binary Search/binarysearch.java
class Solution {

    public int BinarySearch(int[] nums, int target) {

      int left = 0;
      int right = nums.length - 1;

      while (left <= right) {
        int pivot = left + (right - left) / 2;

        if (nums[pivot] == target) return pivot;

        if (target < nums[pivot]) right = pivot - 1;
        
        else left = pivot + 1;
      }
      return -1;
    }

  }

File: Algorithms/Binary Search/binarySearch.js
const binarySearch = (arr, start, end, target) => {
  if (end >= start) {
    mid = parseInt(start + (end - start) / 2);
    if (arr[mid] == target) {
      return mid;
    } else if (arr[mid] > target) {
      return binarySearch(arr, start, mid - 1, target);
    } else {
      return binarySearch(arr, mid + 1, end, target);
    }
  } else {
    return -1;
  }
};
